programs/stake/src/stake_state.rs

```diff
@@ -863,6 +906,22 @@ pub fn redeem_rewards(
     }
 }
 
+// utility function, used by runtime
+pub fn calculate_points(
+    stake_account: &Account,
+    vote_account: &Account,
+    stake_history: Option<&StakeHistory>,
+) -> Result<u128, InstructionError> {
+    if let StakeState::Stake(_meta, stake) = stake_account.state()? {
+        let vote_state: VoteState =
+            StateMut::<VoteStateVersions>::state(vote_account)?.convert_to_current();
```

Comment:
nits: How about extracting this code identical with https://github.com/solana-labs/solana/pull/10914/files#diff-0c1e264efc2089624e55f89513c862b8R888-R890? That would ensure we use identical iteration condition when summing up point and paying rewards. (Yeah, I've tried to find a discrepancy but to no avail..)

Reply:
these functions differ in account mutability...